src/main/java/uk/gov/digital/ho/pttg/application/HmrcClient.java

```diff
@@ -107,7 +116,7 @@ private void storeEmploymentResource(String accessToken, IncomeSummaryContext co
 
     private void storeSelfAssessmentResource(String accessToken, LocalDate fromDate, LocalDate toDate, IncomeSummaryContext context) {
         String toTaxYear = getTaxYear(toDate);
-        String fromTaxYear = getTaxYear(fromDate);
+        String fromTaxYear = getTaxYearForDateOrEarliestAllowed(fromDate, earliestAllowedTaxYear());
```

Comment:
This line is making me squirm but I can't put my finger on it.  Maybe because the method has too many responsibilities (and thus an overly long name)?  Would it be better to just calculate tax year, calculate earliest tax year and taking the maximum of the two?

Reply:
I've gone for this approach now